cloud/gke.go

```diff
@@ -0,0 +1,1038 @@
+	Locations []string
+	// Network
+	Network string
+	// Sub Network
+	SubNetwork string
+	// Configuration for LegacyAbac
+	LegacyAbac bool
+	// NodePool id
+	NodePoolID string
+	// Image Type
+	ImageType string
+}
+
+func waitForCluster(svc *gke.Service, cc GKECluster) (*gke.Cluster, error) {
+
+	message := ""
+	for {
+
+		cluster, err := svc.Projects.Zones.Clusters.Get(cc.ProjectID, cc.Zone, cc.Name).Context(context.TODO()).Do()
+		if err != nil {
+			return nil, err
+		}
+
+		if cluster.Status == statusRunning {
+			banzaiUtils.LogInfof(banzaiConstants.TagCreateCluster, "Cluster %v is running", cc.Name)
+			return cluster, nil
+		}
+
+		if cluster.Status != message {
+			banzaiUtils.LogInfof(banzaiConstants.TagCreateCluster, "%s cluster %s", string(cluster.Status), cc.Name)
+			message = cluster.Status
+		}
+		time.Sleep(time.Second * 5)
+
+	}
+}
+
+func GetClusterGoogle(svc *gke.Service, cc GKECluster) (*gke.Cluster, error) {
+	return svc.Projects.Zones.Clusters.Get(cc.ProjectID, cc.Zone, cc.Name).Context(context.TODO()).Do()
+}
+
+func ReadClusterGoogle(cs *banzaiSimpleTypes.ClusterSimple, svc *gke.Service) *ClusterRepresentation {
+	banzaiUtils.LogInfo(banzaiConstants.TagGetCluster, "Read google cluster with", cs.Name, "id")
+
+	if cs == nil {
+		banzaiUtils.LogInfo(banzaiConstants.TagGetCluster, "<nil> cluster")
+		return nil
+	}
+
+	gkec := GKECluster{
+		ProjectID: cs.Google.Project,
+		Name:      cs.Name,
+		Zone:      cs.Location,
+	}
+
+	response, err := GetClusterGoogle(svc, gkec)
+	if err != nil {
+		banzaiUtils.LogInfo(banzaiConstants.TagGetCluster, "Something went wrong under read:", err)
+		return nil
+	} else {
+		banzaiUtils.LogInfo(banzaiConstants.TagGetCluster, "Read cluster success")
+		clust := ClusterRepresentation{
+			Id:        cs.ID,
+			Name:      cs.Name,
+			CloudType: banzaiConstants.Google,
+			GoogleRepresentation: &GoogleRepresentation{
+				GoogleCluster: response,
+			},
+		}
+		return &clust
+	}
+}
+
+type GoogleRepresentation struct {
+	GoogleCluster *gke.Cluster `json:"value,omitempty"`
+}
+
+func GetClusterInfoGoogle(cs *banzaiSimpleTypes.ClusterSimple, c *gin.Context) {
+	banzaiUtils.LogInfo(banzaiConstants.TagGetCluster, "Fetch aks cluster with name:", cs.Name, "in", cs.Azure.ResourceGroup, "resource group.")
+
+	if cs == nil {
+		banzaiUtils.LogInfo(banzaiConstants.TagGetCluster, "<nil> cluster")
+		SetResponseBodyJson(c, http.StatusInternalServerError, gin.H{
+			JsonKeyStatus: http.StatusInternalServerError,
+		})
+		return
+	}
+
+	svc, err := GetGoogleServiceClient()
+	if err != nil {
+		SetResponseBodyJson(c, http.StatusInternalServerError, gin.H{
+			JsonKeyStatus:  http.StatusInternalServerError,
+			JsonKeyMessage: err,
+		})
+		return
+	}
+
+	gkec := GKECluster{
+		ProjectID: cs.Google.Project,
+		Name:      cs.Name,
+		Zone:      cs.Location,
+	}
+
+	response, err := GetClusterGoogle(svc, gkec)
+	if err != nil {
+		// fetch failed
```

Comment:
Log this, that fetch failed

Reply:
It's logged, but only info:
https://github.com/banzaicloud/pipeline/blob/google-integration/cloud/gke.go#L358
I'll fix it.